### PR TITLE
[docs] review and update bare installation guide for dev-client

### PR DESCRIPTION
**Pull request by esamelson**
# Why

This guide was out of date and structured for older SDK versions. This PR brings the guide up to date, removes some very outdated information, and cleans up the tabs so that there are fewer splits on the page in the default flow.

# How

- Remove the `react-native-unimodules` tabs and setup instructions since they are very out of date at this point.
- Add "Set up Expo modules" as a clear first step since it's now required for dev client.
- Add an additional step to run `npx pod-install` after installation for projects with an `ios` directory.
- Simplify tab titles and move the iOS and Android headers into the tabs (since the SDK 45+ tabs have no need for them).
- Remove the optional step to "Enable development with Expo CLI" since that's already part of setting up Expo modules.
- Update instructions for setting up the expo-updates integration for SDK 45+.

# Test Plan

Ran locally, everything renders as expected:

Screen Shot 2022-09-29 at 2 00 41 PM

Screen Shot 2022-09-29 at 2 00 50 PM

Screen Shot 2022-09-29 at 2 00 58 PM

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/development/installation.md [hunk 1]
```diff
@@ -14,37 +14,32 @@ If you're just starting your project, you can create a new project from our temp
 
 <Terminal cmd={["$ npx crna -t with-dev-client"]} />
 
-If you have an existing project, you'll need to [install the package and make a few changes](installation.md) to your **AppDelegate.m**, **MainActivity.java** and **MainApplication.java**.
+Otherwise, follow this guide to add `expo-dev-client` to your existing React Native or Bare project.
 
-If you created your project with `expo init` or `create-expo-app`, or you already have `expo` and/or other Expo modules up and running, use the tabs marked **With Expo modules**.
+## 0. Set up Expo modules
```
Comment (amandeepmittal) [inline docs/pages/development/installation.md:19]:
```suggestion
## 1. Set up Expo modules
```

As per our [writing style guide here](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md), for consistency reasons, let's start numbered lists with `1` 😄

File: docs/pages/development/installation.md [hunk 2]
```diff
@@ -14,37 +14,32 @@ If you're just starting your project, you can create a new project from our temp
...
 
-If you created your project with `expo init` or `create-expo-app` before SDK 43 and have `react-native-unimodules` up and running, use the tabs marked **With unimodules**. When you're ready, consider upgrading to the newer Expo modules by [following these steps](https://expo.fyi/expo-modules-migration).
+If you created your project with `npx react-native init` and do not have any other Expo packages installed, **you'll need to [install Expo modules](../bare/installing-expo-modules)** before proceeding.
 
-If you created your project with `npx react-native init` and do not have any other Expo packages installed, **starting from version `0.8.0` you'll need to [install Expo modules](../bare/installing-expo-modules)** in order to use `expo-dev-client`.
-
-> **Note**: if you are using _SDK 43 or above_, you'll need to install `expo-dev-client@0.6.0` or above.
+If you created your project with `expo init` or `create-expo-app`, or you already have `expo` and/or other Expo modules up and running, you can skip this step.
 
 ## 1. Installation
```
Comment (amandeepmittal) [inline docs/pages/development/installation.md:25]:
```suggestion
## 2. Installation
```

File: docs/pages/development/installation.md [hunk 3]
```diff
@@ -14,37 +14,32 @@ If you're just starting your project, you can create a new project from our temp
...
 
 <Tabs tabs={["SDK 45+", "SDK below 45"]}>
 
 <Tab >
 
-Make sure that your project is configured to deploy on an iOS version _above 10_.
-To do that, open Xcode and go to General > Deployment Info, and select an iOS version of at least 11.0.
-
-<img src="/static/images/client/check_ios_version.png" style={{maxWidth: "100%" }}/>
+No additional changes are needed to install the package on SDK 45+.
```
Comment (amandeepmittal) [inline docs/pages/development/installation.md:35]:
```suggestion
No additional changes are needed to install the package on SDK 45 and above.
```

As per the [writing style guide here](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#symbols-as-words), let's avoid using symbols.

File: docs/pages/development/installation.md [hunk 4]
```diff
@@ -58,70 +53,48 @@ To do that, open Xcode and go to General > Deployment Info, and select an iOS ve
...
+## 2. Basic configuration
 
-See the [uri-scheme package](https://www.npmjs.com/package/uri-scheme) for more information.
+<Tabs tabs={["SDK 45+", "SDK below 45"]}>
 
-### iOS
+<Tab >
 
-<Tabs tabs={["SDK 45+/expo-modules-core@0.9.1+", "With Expo modules", "With unimodules"]}>
+If you already have your `ios` or `android` directories on disk, you will need to rerun the config plugin in order to configure `expo-dev-client`:
```
Comment (amandeepmittal) [inline docs/pages/development/installation.md:70]:
```suggestion
If you already have your `ios` or `android` directories on disk, you will need to re-run the config plugin to configure `expo-dev-client`:
```

File: docs/pages/development/installation.md [hunk 5]
```diff
@@ -58,70 +53,48 @@ To do that, open Xcode and go to General > Deployment Info, and select an iOS ve
...
 
-Make the following changes to allow the `expo-dev-client` library to control project initialization in the **DEBUG** mode.
+Development builds use deep links to open projects from the QR code. If you have added a custom deep link scheme to your project, your development build will use it. However, if this isn't the case, you need to configure the deep link support for your app. The `uri-scheme` package will do this for you once you have chosen a scheme.
 
-<DiffBlock source="/static/diffs/client/app-delegate-expo-modules.diff" />
-</Tab>
+<InstallSection packageName="expo-development-client" cmd={["npx uri-scheme add <your scheme>"]} hideBareInstructions />
 
-<Tab >
+See the [uri-scheme package](https://www.npmjs.com/package/uri-scheme) for more information.
```
Comment (amandeepmittal) [inline docs/pages/development/installation.md:84]:
```suggestion
For more information, see [uri-scheme package](https://www.npmjs.com/package/uri-scheme).
```

File: docs/pages/development/installation.md [hunk 6]
```diff
@@ -130,12 +103,6 @@ Make the following changes to allow the `expo-dev-client` library to control pro
...
 
-### Enable development with Expo CLI
-
-Expo CLI requires you to have the `expo` package installed so it can maintain consistent behavior in your project when new versions of the Expo SDK are released. The package will not be used directly by your project unless you import it in your application code, which is not recommended.
-
-<InstallSection packageName="expo" cmd={["npm install expo --save-dev"]} hideBareInstructions />
-
 ### Disable packager autostart when building for iOS
 
 When you start your project on iOS, the metro bundler will be started automatically. This behavior might not be ideal when you want to use `npx expo start`. Our recommended solution is to remove the `Start Packager` action from building scripts. To do that you need to open the Xcode, go to "Build Phases" and remove the "Start Packager" action.
```
Comment (amandeepmittal) [inline docs/pages/development/installation.md:108]:
```suggestion
When you start your project on iOS, the metro bundler will start automatically. This behavior might not be ideal when you want to use `npx expo start`. Our recommended solution is to remove the `Start Packager` action from building scripts. To do that you need to open the Xcode, go to "Build Phases" and remove the "Start Packager" action.
```

These suggestions are unrelated to this PR but I think it's a good improvement.

File: docs/pages/development/installation.md [hunk 7]
```diff
@@ -58,70 +53,48 @@ To do that, open Xcode and go to General > Deployment Info, and select an iOS ve
...
 No additional changes are needed to install the package on Android.
 
-## 2. Basic configuration
+</Tab >
 
-Development builds use deep links to open projects from the QR code. If you have added a custom deep link scheme to your project, your development build will use it. However, if this isn't the case, you need to configure the deep link support for your app. The `uri-scheme` package will do this for you once you have chosen a scheme.
+</Tabs >
 
-<InstallSection packageName="expo-development-client" cmd={["npx uri-scheme add <your scheme>"]} hideBareInstructions />
+## 2. Basic configuration
```
Comment (amandeepmittal) [inline docs/pages/development/installation.md:64]:
```suggestion
## 3. Basic configuration
```

Also, can I please ask you to change the numbered list pattern for the rest of the sections? 🙏 

I'd have done it myself but I can't review the unchanged diff 😄

File: docs/pages/development/installation.md [hunk 8]
```diff
@@ -156,7 +123,25 @@ import App from "./App";
...
+
+<Tabs tabs={["SDK 45+", "SDK below 45"]}>
+
+<Tab>
+
+To add this feature, install `expo-updates` in your project:
+
+<Terminal cmd={["$ npx expo install expo-updates"]} />
+
+You may need to rerun the config plugin for `expo-dev-client` after doing so.
```
Comment (amandeepmittal) [inline docs/pages/development/installation.md:136]:
```suggestion
After doing so, you may need to re-run the config plugin for `expo-dev-client`.
```

I don't get the context of why this step is a "may"? Is it like some users have to and some don't if they install `expo-updates`?

**Review by amandeepmittal — CHANGES_REQUESTED** [review]
Thank you for updating this page 🙏 

Left some suggestions.

File: docs/pages/development/installation.md (hunk 8 above)
Comment (esamelson) [reply docs/pages/development/installation.md]: Thanks for making me think about this a little harder -- I realized that this step isn't actually necessary at all! 😅  Rewrote a few other places to reflect this, as well.

**Review by amandeepmittal — APPROVED** [review]

File: docs/pages/development/installation.md (hunk 8 above)
Comment (amandeepmittal) [reply docs/pages/development/installation.md]: Thanks for making these changes 🙏 
Love the new change "SDK 44 and below". I think it is more clear to imply the info docs are conveying there.

**Review by lukmccall — APPROVED** [review]

**Pull request merged**